ML/MLService.cs

```diff
@@ -0,0 +1,208 @@
+
+            //create ml context for training
+            var mlContext = new MLContext(seed: 0);
+            var dataView = mlContext.Data.LoadFromEnumerable(request.Dataset);
+
+            //create test train split
+            DataOperationsCatalog.TrainTestData dataSplitView = mlContext.Data.TrainTestSplit(dataView, testFraction: 0.2);
+            IDataView trainData = dataSplitView.TrainSet;
+            IDataView testData = dataSplitView.TestSet;
+
+            //create pipeline for training
+            IEstimator<ITransformer> mlPipeline = BuildTrainingPipeline(mlContext);
+
+            //train model
+            ITransformer mlModel = TrainModel(mlContext, trainData, mlPipeline);
+
+            response.Finished = DateTime.Now;
+
+            //create chatbee model
+            var chatBeeModel = new ChatbeeModel();
+            chatBeeModel.Model = mlModel;
+            chatBeeModel.DataSet = request.Dataset;
+
+            //save model and return file name
+            var modelFileName = chatBeeModel.SaveModelToFile(mlContext, trainData.Schema, "Data\\Models\\");
+
+            //set filename for response
+            response.ModelName = modelFileName;
+
+            //load if indicated
+            if (request.LoadAfterTraining)
+            {
+                if (LoadedModels.ContainsKey(modelFileName))
+                    LoadedModels.Remove(modelFileName);
+
+                LoadedModels.Add(modelFileName, chatBeeModel);
+            }
+
+            //return model data if requested
+            if (request.ReturnModelData)
+            {
+                response.ModelData = chatBeeModel;
+            }
+
+            //set result
+            response.Result = "Training Completed Successfully";
+
+            //return model file name
+            return response;
+        }
+
+        public ScoringResponse Predict(ScoringRequest request)
+        {
+            var response = new ScoringResponse();
+            //Todo: Enable Exact Match attempt?
+
+
+
+            var modelFileName = request.ModelName;
+            var utterance = request.Utterance;
+
+            MLContext mlContext = new MLContext();
+            ChatbeeModel chatbeeModel;
+
+            //load model if not loaded
+            if (!LoadedModels.ContainsKey(modelFileName))
+            {
+                chatbeeModel = new ChatbeeModel().LoadModelFromFile(mlContext, modelFileName);
+                LoadedModels.Add(modelFileName, chatbeeModel); 
+            }
+
+            ChatbeeModel loadedModel = LoadedModels[modelFileName];
+            chatbeeModel = loadedModel;
+
+
+            //attempt to find any exact matches before going to ml
+            foreach (var item in chatbeeModel.DataSet)
+            {
+                if (item.Utterance == request.Utterance)
+                {
+                    //return for exact match
+                    Output exactMatchOutput = new Output();
```

Comment:
You should add a constructor or simply initialization

Reply:
Syntactic Sugar but fine, I have updated this.